### PR TITLE
[#5] [iOS] [UI] As a user, I can sign in with email and password

### LINKED ISSUE #5: [iOS] [UI] As a user, I can sign in with email and password
## Why

Users must be authenticated before taking or viewing any surveys. Our surveys have access restrictions. Some surveys are targeted to a specific group of users only for a more accurate survey result.

The mobile app should allow users to authenticate with their email and password.

## Acceptance Criteria

- Display the Nimble logo.
- Display the blurred background.
- Display the Email text input.
- Set the placeholder text to Email.
- Display the Password text input.
- Set the placeholder text to Password.
- Mask the password input with the native-OS feature.
- Display the Login button.
- **DO NOT** display the Forgot Password button yet. We will work on this on another separate epic.

## Design

Image

Link: https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A195

### PR DESCRIPTION
**Pull request by suho**
Resolves #5

## What happened 👀

- Added `LoginView` to show the login screen
- Added `UITests` for the login screen

## Insight 📝

`N/A`

## Proof Of Work 📹

<table>
<tr>
	<td>
	<td>
</table>

### CODE REVIEW

**Review by nkhanh44 — COMMENTED** [review]
LGTM, clean UI 💯 . There is just some suggestion.

**Comment by nkhanh44** [comment]
Screen Shot 2022-09-07 at 17 05 08

BTW, in design we notice that there's an overlay with a linear gradient on background image. I think if we apply this feature, it's going to be more identical to the design 😄 . Anyway, it's optional and up to you 👍 .

**Comment by nkhanh44** [comment]
Please provide us POW of textfield's placeholder too 👍 .

File: ios/Surveys/Sources/Presentation/Views/PrimaryButton/PrimaryButton.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    private let title: String
+    private let action: () -> Void
+
+    var body: some View {
+        GeometryReader { geometry in
+            Button(action: action) {
+                Text(title)
+                    .frame(height: geometry.size.height, alignment: .center)
+                    .font(.system(size: 17.0, weight: .bold))
```
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Views/PrimaryButton/PrimaryButton.swift:21]: Good catch!

File: ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    @State var animated = false
+    @State private var email: String = ""
+    @State private var password: String = ""
+
+    var body: some View {
+        ZStack {
+            R.image.splashBackground.image
+                .resizable()
+                .blur(radius: animated ? 20 : 0)
```
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift:21]: Good catch!

**Comment by suho** [comment]
> Screen Shot 2022-09-07 at 17 05 08
> 
> BTW, in design we notice that there's an overlay with a linear gradient on background image. I think if we apply this feature, it's going to be more identical to the design 😄 . Anyway, it's optional and up to you 👍 .

@nkhanh44 Actually, I'm using a picture with gradient already, so we don't need to write grandient code for this screen

File: ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift]: Fixed in https://github.com/suho/kmm-ic/pull/81/commits/c3e422d1d1fd71fce55d6d56938fb5915976d772

File: ios/Surveys/Sources/Presentation/Views/PrimaryButton/PrimaryButton.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Views/PrimaryButton/PrimaryButton.swift]: Fixed in https://github.com/suho/kmm-ic/pull/81/commits/c3e422d1d1fd71fce55d6d56938fb5915976d772

**Review by nkhanh44 — APPROVED** [review]

**Review by edgarss — APPROVED** [review]

File: ios/Surveys/Resources/Assets/Assets.xcassets/splashBackground.imageset/Contents.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "images" : [
     {
-      "filename" : "Background.pdf",
+      "filename" : "Image (1).pdf",
```
Comment (vnntsu) [inline ios/Surveys/Resources/Assets/Assets.xcassets/splashBackground.imageset/Contents.json:4]: Nit: the name of the image file is a little bit weird, I think we can choose a legal one.

File: ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+struct LoginView: View {
+
+    @State private var animated = false
+    @State private var email: String = ""
+    @State private var password: String = ""
+
+    var body: some View {
+        ZStack {
+            R.image.splashBackground.image
```
Comment (vnntsu) [inline ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift:19]: I suggest creating a few `typealiases` for RSwift:

```
typealias Font = R.font
typealias Localize = R.string.localizable
typealias Asset = R.image
```

File: ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                .resizable()
+                .blur(radius: animated ? 20.0 : 0.0)
+                .animation(.easeInOut(duration: 0.5))
+
+            R.image.logoWhite.image
+                .offset(x: 0.0, y: animated ? -229.0 : 0.0)
+                .scaleEffect(animated ? (1.0 / 1.2) : 1.0)
+                .animation(.easeInOut(duration: 0.5))
+                .onAppear {
```
Comment (vnntsu) [inline ios/Surveys/Sources/Presentation/Modules/Login/LoginView.swift:28]: Just a concern, what do you think if we move this `onAppear` to `ZStack`?

**Review by vnntsu — COMMENTED** [review]
There is a couple of minor comments. The rest is looking good!

File: ios/Surveys/Resources/Localizations/en.lproj/Localizable.strings [hunk 6]
```diff
@@ -0,0 +1,3 @@
+"authentication.fields.email" = "Email";
+"authentication.fields.password" = "Password";
+"authentication.fields.login" = "Log in";
```
Comment (blyscuit) [inline ios/Surveys/Resources/Localizations/en.lproj/Localizable.strings:3]: should this be button?
```suggestion
"authentication.button.login" = "Log in";
```

**Review by vnntsu — APPROVED** [review]
LGTM! 🚀

File: ios/Surveys/Resources/Localizations/en.lproj/Localizable.strings (hunk 6 above)
Comment (suho) [reply ios/Surveys/Resources/Localizations/en.lproj/Localizable.strings]: Fixed in https://github.com/suho/kmm-ic/pull/81/commits/9e06a002678aacd2bd2b219835f3f7c5a070f1f6

**Pull request merged**